.typedoc/custom-plugin.mjs

```diff
@@ -34,16 +36,37 @@ const LINK_REPLACEMENTS = [
  * It also shouldn't matter how level deep the relative link is.
  *
  * This function returns an array of `{ pattern: string, replace: string }` to pass into the `typedoc-plugin-replace-text` plugin.
+ *
+ * @example
+ * [foo](../../bar.mdx) -> [foo](/new-path)
+ * [foo](./bar.mdx) -> [foo](/new-path)
+ * [foo](bar.mdx) -> [foo](/new-path)
+ * [foo](bar.mdx#some-id) -> [foo](/new-path#some-id)
  */
 function getRelativeLinkReplacements() {
   return LINK_REPLACEMENTS.map(([fileName, newPath]) => {
     return {
-      pattern: new RegExp(`\\((?:\\.{1,2}\\/)+.*?${fileName}\\.mdx\\)`, 'g'),
-      replace: `(${newPath})`,
+      // Match both path and optional anchor
+      pattern: new RegExp(`\\((?:(?:\\.{1,2}\\/)+[^()]*?|)${fileName}\\.mdx(#[^)]+)?\\)`, 'g'),
```

Comment:
The new regex accounts for two bugs I saw:

- A string contained multiple `[]()` and previously it replaced parts outside of that link (until the next occurence)
- `[foo](bar.mdx)` and `[foo](bar.mdx#test)` didn't work